stable/cluster-autoscaler/templates/deployment.yaml

```
@@ -68,6 +68,20 @@ spec:
           {{- if and (eq .Values.cloudProvider "aws") (ne .Values.awsRegion "") }}
             - name: AWS_REGION
               value: "{{ .Values.awsRegion }}"
+            {{- if not ( empty .Values.awsAccessKeyId) }}
```

I think this should be enough:

```suggestion
            {{- if .Values.awsAccessKeyID }}
```